RIOT-OS/RIOT
# boards/cc2538: add riotboot feature

### Contribution description

This PR adds riotboot to cc2538. This is a first step for this since there is a some handling of the `Flash Customer Configuration Area (CCA)` that should be improved.

The CCA holds among other configurations the vector table entrypoint so it is important that this always points the bootloader, this means that it should idealy:

- never be overriden when building only riotboot `slots`
- be written when writing bootloader, extendended or combined bin's

Currently I'm no writing the CCA whenever `riotboot` is used, that means that it must have been written previously so the cc2538 device needs to have already been flashed with any other riot application.

I would like to fix this by writing the CCA independently, but I'll address in a follow up.

Currently flashing only will work properly with jlink. I need to make upstream changes to `cc2538.bsl` so I'll add this in a follow up as well.

Initial work on this was done by @brent7984 in #11665, I've kept him as co-author.

### Testing procedure

- run `tests/riotboot` on a cc2538 board

I ran `examples/suit_update `successfully:

`PROGRAMMER=jlink BOARD=openmote-b make -C examples/suit_update/ flash test -j3`

```
iotboot_flashwrite: initializing update to target slot 0
Fetching firmware |█████████████████████████| 100%
Finalizing payload store
Verifying image digest
Starting digest verification against image
Install correct payload
Verifying image digest
Starting digest verification against image
Install correct payload
Image magic_number: 0x544f4952
Image Version: 0x5fa52bcc
Image start address: 0x00201400
Header chksum: 0x53bb3d33
suit_coap: rebooting...
----> ethos: hello received
Failed to send flush request: Operation not permitted
gnrc_uhcpc: Using 4 as border interface and 0 as wireless interface.
gnrc_uhcpc: only one interface found, skipping setup.
main(): This is RIOT! (Version: 2021.01-devel-622-g19aed-wip/cc2538_riotboot)
RIOT SUIT update example application
running from slot 0
slot start addr = 0x201000
Image magic_number: 0x544f4952
Image Version: 0x5fa52bcc
Image start address: 0x00201400
Header chksum: 0x53bb3d33
suit_coap: started.
Starting the shell
> ifconfig
ifconfig
Iface  4  HWaddr: AE:8B:6E:51:70:9C
          L2-PDU:1500  MTU:1500  HL:64  RTR
          Source address length: 6
          Link type: wired
          inet6 addr: fe80::ac8b:6eff:fe51:709c  scope: link  VALpinging node...
PING fe80::ac8b:6eff:fe51:709c%riot0(fe80::ac8b:6eff:fe51:709c%riot0) 56 data bytes
--- fe80::ac8b:6eff:fe51:709c%riot0 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 60.024/60.024/60.024/0.000 ms
pinging node succeeded.
TEST PASSED
``` 

### Issues/PRs references

Depends on #15387
Closes #11665
